src/rgw/services/svc_mdlog.cc

```diff
@@ -49,7 +49,7 @@ int RGWSI_MDLog::do_start(optional_yield y, const DoutPrefixProvider *dpp)
                                             current_period));
 
   if (run_sync &&
-      svc.zone->need_to_sync()) {
+      !current_period.get_id().empty()) {
```

Comment:
i'm assuming that `run_sync` is coming from `rgw_run_sync_thread` - we don't want this to be conditional on that anymore, in order to resolve https://tracker.ceph.com/issues/49615

Reply:
Is it correct to get rid of svc.zone->need_to_sync() as well? Just want to make sure both conditions should be got rid of and replaced.

Reply:
i think this can just be `if (svc.zone->need_to_sync()) {`. need_to_sync() already checks that the period isn't empty, along with some other stuff that doesn't appear to be conditional on rgw_run_sync_thread

Reply:
All right, fixed.